doc/api/url.md

```diff
@@ -250,7 +250,464 @@ properties of URL objects:
+is returned as a String.
+
+```js
+const myURL = new URL('http://example.org:8888');
+console.log(myURL.port);
+  // Prints 8888
+
+myURL.port = 1234;
+console.log(myURL.href);
+  // Prints http://example.org:1234
+```
+
+The port value may be set as either a number or as a String containing a number
+in the range `0` to `65535` (inclusive). Setting the value to the default port
+of the `URL` objects given `protocol` will result in the `port` value becoming
+the empty string (`''`).
+
+Using the `delete` keyword (e.g. `delete myURL.port`) has no effect but will
+still return `true`.
+
+Invalid URL port values assigned to the `port` property are ignored.
+
+#### url.protocol
+
+Gets and sets the protocol portion of the URL.
+
+```js
+const myURL = new URL('http://example.org');
+console.log(myURL.protocol);
+  // Prints http:
+
+myURL.protocol = 'ftp';
+console.log(myURL.href);
+  // Prints ftp://example.org
+```
+
+Using the `delete` keyword (e.g. `delete myURL.protocol`) has no effect but will
+still return `true`.
+
+Invalid URL protocol values assigned to the `protocol` property are ignored.
+
+#### url.search
+
+Gets and sets the serialized querystring portion of the URL.
+
+```js
+const myURL = new URL('http://example.org/abc?123');
+console.log(myURL.search);
+  // Prints ?123
+
+myURL.search = 'abc=xyz';
+console.log(myURL.href);
+  // Prints http://example.org/abc?abc=xyz
+```
+
+Using the `delete` keyword (e.g. `delete myURL.search`) has no effect but will
+still return `true`.
+
+Any invalid URL characters appearing in the value assigned the `search`
+property will be [percent-encoded](#whatwg-percent-encoding). Note that the
+selection of which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.searchParams
+
+Gets a [`URLSearchParams`](#url_class_urlsearchparams) object representing the
+querystring parameters of the URL.
+
+#### url.username
+
+Gets and sets the username portion of the URL.
+
+```js
+const myURL = new URL('http://abc:xyz@example.com');
+console.log(myURL.username);
+  // Prints abc
+
+myURL.username = '123';
+console.log(myURL.href);
+  // Prints http://123:xyz@example.com
+```
+
+Using the `delete` keyword (e.g. `delete myURL.username`) has no effect but will
+still return `true`.
+
+Any invalid URL characters appearing in the value assigned the `username`
+property will be [percent-encoded](#whatwg-percent-encoding). Note that the
+selection of which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.toString([unicode])
+
+* `unicode` {Boolean} If `true`, Unicode characters within the hostname will
+  be encoded as-is without [Punycode][] encoding. Defaults to `false`.
+
+The `toString()` method on the `URL` object returns the serialized URL. When
+`unicode` is `false`, the value returned is equivalent to that of `url.href`.
```

Comment:
the [unicode] extension appears nonstandard. It should be removed, or at least not documented?